### PR TITLE
feat(menus): standardized usage of entity menu and added a social menu

### LINKED ISSUE #5707: Split entity menu items in primaire and secundairy and introduce a dropdown
They both use the X icon, which makes it easy to confuse if you're hiding from the river or deleting the entity completely.

### LINKED ISSUE #6936: Display entity menu by default in the object/default view
This line https://github.com/Elgg/Elgg/blob/1.9/views/default/object/default.php#L22 is trying to add a non-existing view `'navigation/menu/metadata'` into the metadata part of the view.

There are no references to this view anywhere in the core, so it's either a mistake or an undocumented view extension point.

In any case I think we should replace it with entity menu:

``` php
$metadata = elgg_view_menu('entity', array(
    'entity' => $vars['entity'],
    'handler' => $vars['entity'->subtype,
    'sort_by' => 'priority',
    'class' => 'elgg-menu-hz',
));
```

This would make it much faster to get started when working on adding new object subtypes.

### LINKED ISSUE #9977: Wrap summary listing metadata and stop floating entity menu
https://github.com/Elgg/Elgg/pull/9944#discussion_r70045689

### LINKED ISSUE #10025: User expectation is not clear deleting a river item
Users may reasonably expect the referenced object to be deleted, particularly for things like wire posts.

Strawman: river delete button opens a dialog: "Note that removing this activity item will not affect the site content that it references:" (show short view of object). 

[Remove activity item] cancel

### PR DESCRIPTION
**Pull request by jdalsem**
Fixes #6936
Fixes #10025
Fixes #9977
Fixes #5707 

**River Menu**
image

**Entity Menu**
image

### CODE REVIEW

File: mod/groups/start.php [hunk 1]
```diff
@@ -348,47 +351,36 @@ function groups_activity_owner_block_menu($hook, $type, $return, $params) {
...
+		return;
 	}
+	
+	$isFeatured = $entity->featured_group == "yes";
 
+	$return[] = ElggMenuItem::factory([
+		'name' => 'feature',
+		'icon' => 'arrow-up',
+		'text' => elgg_echo("groups:makefeatured"),
+		'href' => elgg_add_action_tokens_to_url("action/groups/featured?group_guid={$entity->guid}&action_type=feature"),
```
Comment (jeabakker) [inline mod/groups/start.php]: setting `is_action` in the menu item will add the tokens

File: mod/groups/start.php [hunk 2]
```diff
@@ -348,47 +351,36 @@ function groups_activity_owner_block_menu($hook, $type, $return, $params) {
...
-			'priority' => 300,
-			'item_class' => $isFeatured ? '' : 'hidden',
-			'data-toggle' => 'feature',
-		]);
-	}
+	$return[] = ElggMenuItem::factory([
+		'name' => 'unfeature',
+		'icon' => 'arrow-down',
+		'text' => elgg_echo("groups:makeunfeatured"),
+		'href' => elgg_add_action_tokens_to_url("action/groups/featured?group_guid={$entity->guid}&action_type=unfeature"),
```
Comment (jeabakker) [inline mod/groups/start.php]: setting `is_action` in the menu item will add the tokens

File: mod/groups/views/default/groups/imprint/member_count.php [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+$entity = elgg_extract('entity', $vars);
+if (!($entity instanceof \ElggGroup)) {
+	return;
+}
+
+// number of members
+$content = $entity->getMembers(['count' => true]) . ' ' . elgg_echo('groups:member');
+
+$subtitle .= elgg_format_element('div', ['class' => 'groups-description'], $entity->briefdescription);
```
Comment (jeabakker) [inline mod/groups/views/default/groups/imprint/member_count.php]: should be `$content`

File: mod/likes/start.php [hunk 4]
```diff
@@ -152,16 +157,15 @@ function likes_river_menu_setup($hook, $type, $return, $params) {
...
 
 	return ElggMenuItem::factory([
 		'name' => 'likes',
 		'href' => '#',
-		'text' => elgg_view_icon('thumbs-up', [
-			'class' => $is_liked ? 'elgg-state-active' : '',
-		]),
-		'title' => elgg_echo($is_liked ? 'likes:remove' : 'likes:likethis'),
+		'icon' => 'thumbs-up',
+		'class' =>  $is_liked ? 'elgg-state-active' : '',
```
Comment (jeabakker) [inline mod/likes/start.php]: is `class` supported?

File: mod/pages/start.php [hunk 5]
```diff
@@ -229,14 +229,14 @@ function pages_owner_block_menu($hook, $type, $return, $params) {
  */
 function pages_entity_menu_setup($hook, $type, $return, $params) {
 	if (elgg_in_context('widgets')) {
-		return $return;
+		return;
 	}
 
 	elgg_load_library('elgg:pages');
 	$entity = $params['entity'];
```
Comment (jeabakker) [inline mod/pages/start.php]: use `elgg_extract`

File: mod/thewire/views/default/object/thewire.php [hunk 6]
```diff
@@ -6,7 +6,7 @@
  */
 
 $post = elgg_extract('entity', $vars, false);
```
Comment (jeabakker) [inline mod/thewire/views/default/object/thewire.php]: why default to `false`

File: views/default/object/elements/full/responses.php [hunk 7]
```diff
@@ -6,7 +6,17 @@
...
 
-$responses = elgg_extract('responses', $vars);
+if (!elgg_extract('show_responses', $vars, true)) {
+	return;
+}
+
+$entity = elgg_extract('entity', $vars);
+
+$responses = elgg_extract('responses', $vars, null);
+if ($responses === null && $entity) {
```
Comment (jeabakker) [inline views/default/object/elements/full/responses.php]: make sure `$entity` is an `ElggEntity`

File: views/default/object/elements/summary/metadata.php [hunk 8]
```diff
@@ -7,6 +7,23 @@
 
 $metadata = elgg_extract('metadata', $vars);
 if (!$metadata) {
```
Comment (jeabakker) [inline views/default/object/elements/summary/metadata.php]: maybe check if `$metadata === null`

File: views/default/object/elements/summary/subtitle.php [hunk 9]
```diff
@@ -5,7 +5,15 @@
  */
 $subtitle = elgg_extract('subtitle', $vars);
 if (!$subtitle) {
```
Comment (jeabakker) [inline views/default/object/elements/summary/subtitle.php]: maybe check if `$subtitle === null`

**Review by jeabakker — CHANGES_REQUESTED** [review]

File: views/default/object/comment.php [hunk 10]
```diff
@@ -20,61 +22,49 @@
...
-	$anchor = "<a name=\"comment-{$comment->getGUID()}\"></a>";
-
-	$menu = elgg_view_menu('entity', [
-		'entity' => $comment,
-		'handler' => 'comment',
-		'sort_by' => 'priority',
-		'class' => 'elgg-menu-hz float-alt',
-	]);
 	
-	if (elgg_in_context('activity')) {
```
Comment (jeabakker) [inline views/default/object/comment.php]: this is still needed

**Review by jeabakker — CHANGES_REQUESTED** [review]

File: mod/pages/views/default/object/page_top.php [hunk 11]
```diff
@@ -15,7 +15,7 @@
 $revision = elgg_extract('revision', $vars, false);
 
 if (!$page) {
```
[the remark this answers is not in the document]
Comment (jdalsem) [reply mod/pages/views/default/object/page_top.php]: no, feel free to PR :)

File: views/default/object/comment.php (hunk 10 above)
Comment (jdalsem) [reply views/default/object/comment.php]: fixed

**Review by jeabakker — APPROVED** [review]

File: engine/lib/comments.php [hunk 12]
```diff
@@ -431,6 +435,48 @@ function _elgg_comments_prepare_notification($hook, $type, $returnvalue, $params
 	return $returnvalue;
 }
 
+/**
+ * Adds comment menu items to entity menu
+ *
```
Comment (hypeJunction) [inline engine/lib/comments.php]: Could you please make a habit of documenting the hook name. It's impossible to read the code without an IDE

Comment (jdalsem) [reply engine/lib/comments.php]: good point

**Pull request merged**